wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/modal/ModalDialog.css

```diff
@@ -0,0 +1,43 @@
+body.modal-dialog-open {
+	
+}
+
+body.modal-dialog-no-scroll {
+	overflow: hidden;
+	height: 100%;
+	width: 100%;
+}
+
+body.modal-dialog-open-ios {
+	position: fixed;
+}
+
+.modal-dialog-overlay {
+	position: fixed;
+	top: 0;
+	left: 0;
+	width: 100%;
+	height: 100%;
+	margin: 0;
+	padding: 0;
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	flex-direction: column;
+	z-index: 1000000;
+}
+
+.modal-dialog {
+	position: relative;
+	width: 100%;
+	margin: auto;
+	overflow: hidden;
+	z-index: 1000010;
```

Comment:
or may be this one should be configurable ? Or both ?!

Reply:
neither should be configurable from java imho, they will be overridden by application-specific css. eg we have this in our css:
```
.modal-dialog-overlay {
	z-index:3 !important;
	& ~ .select2-drop-mask {
		z-index:3;
	}
	& ~ .select2-drop {
		z-index:4; //make sure select2 dropdown is visible inside a modal
	}
}
```